### PR TITLE
Add docs for pagination

### PR DESCRIPTION
**Pull request by chloepouprom**
@jpjoyal @alavers @mspensieri @jugarrit @dannytranlx

### CODE REVIEW

File: source/rest/index.md [hunk 1]
```diff
@@ -1104,16 +1106,16 @@ Removes the specified channel from the appUser's clients.
...
 When the first message is sent to an app user or received from an app user, a conversation is automatically created for them. The conversation and messages for a given app user can be retrieved and created by way of the `/v1/appusers/` API.
 
-## Get Conversation
+## Get Messages
 
 > Request:
 
 ```shell
-curl https://api.smooch.io/v1/appusers/c7f6e6d6c3a637261bd9656f/conversation \
+curl https://api.smooch.io/v1/appUsers/c7f6e6d6c3a637261bd9656f/messages?before=57bb31808f75d3c3948b8500 \
```
Comment (mspensieri) [inline source/rest/index.md]: Should change all these to use timestamps instead of ids

File: source/rest/index.md [hunk 2]
```diff
@@ -1129,23 +1131,34 @@ smooch.conversations.get('c7f6e6d6c3a637261bd9656f').then((response) => {
     "_id": "df0ebe56cbeab98589b8bfa7",
     "appMakers": [],
```
Comment (mspensieri) [inline source/rest/index.md]: don't think `appMakers` and `appUsers` are part of the response anymore

File: source/rest/index.md [hunk 3]
```diff
@@ -68,6 +68,8 @@ Smooch APIs are subject to rate limiting.  If you exceed the limits, Smooch will
 
 The "devices" array returned in the AppUser payload has been changed to "clients". In order to maintain compatibility, we will keep returning the "devices" array in v1, but in the next version it will be removed.
 
+The endpoints for [retrieving a conversation](#get-conversation), [posting a message](#post-message), and [uploading an image](#upload-image) have been changed. To maintain compatibility, both the new and old endpoint will be usable. However, the old endpoint will be removed in the next version.
```
Comment (mspensieri) [inline source/rest/index.md]: I wonder if we should keep the old endpoints around anyways, but just don't document any of the parameters?

```
<api>`POST /v1/appusers/{smoochId|userId}/conversation/messages`</api>
This API is deprecated, use #post-message instead
```

File: source/rest/index.md [hunk 4]
```diff
@@ -1129,23 +1131,34 @@ smooch.conversations.get('c7f6e6d6c3a637261bd9656f').then((response) => {
...
+    "avatarUrl": "https://www.gravatar.com/image.jpg",
+    "received": 1439220041.586
+  }]
 }
 ```
 
-<api>`GET /v1/appusers/{smoochId|userId}/conversation`</api>
+<api>`GET /v1/appusers/{smoochId|userId}/messages`</api>
+
+Get the specified app user's conversation history with a limit of 100 messages, if it exists. If the conversation has not yet been created for the specified app user, 404 will be returned.
```
Comment (mspensieri) [inline source/rest/index.md]: nit: If _a_ conversation has not yet been created

File: source/rest/index.md [hunk 5]
```diff
@@ -1129,23 +1131,34 @@ smooch.conversations.get('c7f6e6d6c3a637261bd9656f').then((response) => {
...
+
+### Pagination
+
+The API endpoint for retrieving messages of a conversation has a limit of a 100 messages. The `before` and `after` parameters will have to be specified to indicate which range of messages to return. These paramaters are mutually exclusive. If neither is specified, then the most recent 100 messages will be returned.
 
-Get the specified app user's conversation history, if it exists. If the conversation has not yet been created for the specified app user 404 will be returned.
+| Parameter                | Description              |
+|--------------------------|--------------------------|
+| `before`                 | Timestamp of message. The API will return 100 messages before the specified timestamp.           |
+| ` after`                 | Timestamp of message. The API will return 100 messages after the specified timestamp.            |
```
Comment (mspensieri) [inline source/rest/index.md]: extraneous space before `after`.

Also, should we mention that the specified time range has an exclusive boundary? i.e. a message with the provided timestamp will _not_ be returned

File: source/rest/index.md [hunk 6]
```diff
@@ -1129,23 +1131,34 @@ smooch.conversations.get('c7f6e6d6c3a637261bd9656f').then((response) => {
...
+
+The timestamp format should be in [Unix time](https://en.wikipedia.org/wiki/Unix_time). Note that in the old endpoint `/v1/appusers/{smoochId|userId}/messages`, pagination was not supported.
```
Comment (mspensieri) [inline source/rest/index.md]: Don't think we need to call out the old endpoint here

File: source/rest/index.md [hunk 7]
```diff
@@ -1279,14 +1283,14 @@ There are 4 types of supported actions : **link**, **buy**, **postback**, and **
...
-curl https://api.smooch.io/v1/appusers/c7f6e6d6c3a637261bd9656f/conversation/messages \
+curl https://api.smooch.io/v1/appusers/c7f6e6d6c3a637261bd9656f/messages \
      -X POST \
      -d '{"text":"Just put some vinegar on it", "role": "appMaker", "actions": [{"type": "link", "text": "Put vinegar", "uri": "http://example.com" }]}' \
      -H 'content-type: application/json' \
      -H 'authorization: Bearer your-jwt'
 ```
 ```js
-smooch.conversations.sendMessage('c7f6e6d6c3a637261bd9656f', {
+smooch.sendMessage('c7f6e6d6c3a637261bd9656f', {
```
Comment (mspensieri) [inline source/rest/index.md]: `smooch.appUsers.sendMessage`? Couple places below that have the same error also

File: source/rest/index.md (hunk 2 above)
Comment (chloepouprom) [inline source/rest/index.md]: They're still part of the response

File: source/rest/index.md (hunk 3 above)
Comment (chloepouprom) [inline source/rest/index.md]: Hmm so would there still be a section for the old APIs? Or would I just like the old endpoints in this section?

**Comment by jugarrit** [comment]
👍

File: source/rest/index.md (hunk 3 above)
Comment (jpjoyal) [inline source/rest/index.md]: I'd rather remove them. A compromise might be to have a deprecated section for these

File: source/rest/index.md [hunk 8]
```diff
@@ -1127,25 +1129,35 @@ smooch.conversations.get('c7f6e6d6c3a637261bd9656f').then((response) => {
...
 
-Get the specified app user's conversation history, if it exists. If the conversation has not yet been created for the specified app user 404 will be returned.
+The API endpoint for retrieving messages of a conversation has a limit of a 100 messages. The `before` and `after` parameters will have to be specified to indicate which range of messages to return. These paramaters are mutually exclusive. If neither is specified, then the most recent 100 messages will be returned.
+
+| Parameter                | Description              |
+|--------------------------|--------------------------|
+| `before`                 | Timestamp of message. The API will return 100 messages before the specified timestamp (excluding the message with the provided timestamp).           |
+| `after`                  | Timestamp of message. The API will return 100 messages after the specified timestamp (excluding the message with the provided timestamp).            |
+
+The timestamp format should be in [Unix time](https://en.wikipedia.org/wiki/Unix_time).
```
Comment (jpjoyal) [inline source/rest/index.md]: I think we should be precise about the format here, especially in regards to decimals

Comment (lemieux) [inline source/rest/index.md]: Fixed it.

> The timestamp format should be in seconds using [Unix time](https://en.wikipedia.org/wiki/Unix_time). Note that you can specify milliseconds using a decimal number if needed.

File: source/rest/index.md (hunk 3 above)
Comment (lemieux) [inline source/rest/index.md]: Removed.

**Comment by mspensieri** [comment]
👍

**Pull request merged**